babi.py

```diff
@@ -1410,10 +1424,31 @@ def sanitanize(x):
                             convert_none=False):
                         continue
                 vals = ['now()', str(uid)]
-                vals += [sanitanize(babi_eval(x[0], record, convert_none=x[1]))
-                    for x in dimension_expressions]
-                vals += [sanitanize(babi_eval(x, record, convert_none='zero'))
-                    for x in measure_expressions]
+                for x in dimension_expressions:
+                    try:
+                        vals.append(sanitanize(babi_eval(x[0], record,
+                            convert_none=x[1])))
+                    except Exception as message:
+                        if self.report.babi_raise_user_error:
+                            raise UserError(gettext(
+                                'babi.create_data_exception',
+                                _type='Dimensions',
```

Comment:
You must create two different messages: one for dimensions and another for measures, otherwise you won't be able to translate those "type_" strings.